crenel_caves.logic

```diff
@@ -1,17 +1,118 @@
-area CrenelCaves:
-  room BlocksChest
-  room BlocksPuzzle
-  room BlocksBomb
-  room FloodBlock
-  room FountHP
-  room FountRupee
-  room GraybladeEntrance
-  room Guru
-  room HelmaHall
-  room MushroomGap
-  room RockPush
-  room ScrubBomb
-  room ScrubGrip
-  room ScrubHint
-  room SwitchBridge
-  room PotCircle
\ No newline at end of file
+area CrenelCaves:
+  room BlocksChest:
+    node BlocksChest:
+      door Staircase
+      door Doorway
+      enemy RedChus
+      chest BlocksChest
+  room BlocksPuzzle:
+    node BlocksPuzzle:
+      door Ladder
+      door Doorway
+      enemy Helmasaurs
+    node MushroomGapSouth:
+      door Doorway
+      enemy Helmasaur
+    node MushroomGapNorth:
+      door Staircase
+      enemy Helmasaur
+    MushroomGapNorth <-> MushroomGapSouth: (item GustJar | item Cape | item LightArrows)
+  room BlocksBomb:
+    node PreBombBlocks:
+      door PreBombBlocks
+      enemy SilverChu
+      enemy RedChu
+    node PostBombBlocks:
+      door PostBombBlocks
+      enemy SilverChu
+    PreBombBlocks -> PostBombBlocks: item BombBag
+  room FloodBlock: #This is the room w/ 2 chests and a HP on the very left side of crenel right?
+    node FloodBlockEntrance:
+      door FloodBlock
+    node FloodBlockChests:
+      chest LeftFloodBlockChest
+      chest RightFloodBlockChest
+    FloodBlockEntrance -> FloodBlockChests: (item BombBag | item Cape)
+    node FloodBlockHP:
+      chest HearPiece
+    FloodBlockEntrance -> FloodBlockHP: (item BombBag | item Cape | item Flippers)
+  room FountHP:
+    node FountHP:
+      door FountHP
+      chest HeartPiece
+  room FountRupee:
+    node FountRupee:
+      door FountRupee
+    node FountRupeeItems:
+      chest LeftRupee
+      chest CenterRupee
+      chest RightRupee
+    FountRupee -> FountRupeeItems: setting Rupeemania
+  room GraybladeEntrance:
+    node GraybladeEntrance:
+      door Entrance
+    node Switches:
+	  flag Switches
+    #Completely guessing here on the actual item names/syntax
+	GraybladeEntrance -> Switches: (item SpinAttack & (item RedSword | item BlueSword | item FourSword))
+    node SwitchDoor:
+      door SwitchDoor
+    Switches <-> SwitchDoor
+    GraybladeEntrance -> SwitchDoor: flag SwitchesPressed

```

Comment:
And depending on the changes to the above, this might need to go both ways (if you get put behind the locked door, i.e. can go back the way you came but can't press the switches if coming from the top)

Reply:
Tested this with debug mode, warping to Grayblade and leaving puts you just in front of the door letting you leave even if the door is closed

Reply:
Alright, so the `door` needs to be split into an `entrance` (requiring the flag to get to, on its own node) and `exit` (on the Main node)

Reply:
You use `Switches` for the flag above (where it's set)
```suggestion
    GraybladeEntrance -> SwitchDoor: flag Switches
```